### PR TITLE
Add a Gitlab artifact file handler

### PR DESCRIPTION
**Pull request by Wuestengecko**
This file handler uses the Gitlab API to request files from the "artifacts" archive of a recent successful CI job.

### CODE REVIEW

File: capellambse/filehandler/gitlab_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+                if i >= stop:
+                    return
+
+            match = RE_LINK_NEXT.fullmatch(response.headers["Link"])
+            if not match:
+                break
+            next_url = match.group(0)
+
+    def get_model_info(self) -> loader.ModelInfo:
+        return loader.ModelInfo(branch=self.__branch, url=self.__path)
```
Comment (ewuerger) [inline capellambse/filehandler/gitlab_artifacts.py:241]: There is no docstring. But also, what can you write here that isn't already covered by its name.

File: capellambse/filehandler/gitlab_artifacts.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+    def open(
+        self,
+        filename: str | pathlib.PurePosixPath,
+        mode: t.Literal["r", "rb", "w", "wb"] = "rb",
+    ) -> t.BinaryIO:
+        path = helpers.normalize_pure_path(filename, base=self.subdir)
```
Comment (ewuerger) [inline capellambse/filehandler/gitlab_artifacts.py:248]: No docstring, but also open says it all. Pedantics would ask about a `Raises` section.

**Review by ewuerger — APPROVED** [review]
LGTM.

File: capellambse/filehandler/gitlab_artifacts.py (hunk 2 above)
Comment (Wuestengecko) [reply capellambse/filehandler/gitlab_artifacts.py:248]: The docstring is inherited from the base class.

**Pull request merged**